src/merkle/store.rs

```diff
@@ -284,16 +301,18 @@ impl MerkleStore {
         })
     }
 
+    // DATA MUTATORS
+    // --------------------------------------------------------------------------------------------
+
     pub fn set_node(
         &mut self,
         root: Word,
         index: NodeIndex,
         value: Word,
     ) -> Result<Word, MerkleError> {
-        let current_node = self.get_node(root, index)?;
-        let path = self.get_path(root, index)?;
-        if current_node != value {
-            self.add_merkle_path(index.value(), value, path)
+        let result = self.get_path(root, index)?;
+        if result.0 != value {
+            self.add_merkle_path(index.value(), result.0, result.1)
```

Comment:
Let's add doc comments to this function.

Also, shouldn't line 315 be:
```Rust
self.add_merkle_path(index.value(), value, result.1)
```
If so, should we add a test to make sure this is checked?